### ISSUE TITLE
Msys2 git create strange directory when path with "/" delimiter is used

### ISSUE THREAD

**Issue by zdm** [Alexpux/MSYS2-packages#554]
Please, look at screenshot

![screenshot 2016-04-15 16 50 31](https://cloud.githubusercontent.com/assets/1025608/14563489/e7784f0e-032a-11e6-8423-59640b9303d5.png)

Git created garbage folder "c\" in the current directory.
This is only appeared, when I use unix-style delimiter in path.

**Comment by Alexpux** [comment]
Seems you use not MSYS2 version of git

**Comment by mati865** [comment]
Also don't use it in cmd, but in MSYS2 shell or MinGW shell instead.

**Comment by Alexpux** [comment]
Chech that you use git from my msys2 git package

**Comment by zdm** [comment]
I have only msys2 git in my system.

Running from msys2 shell has the same effect. Garbage folder still create in current working dir.

**Comment by maoueh** [comment]
I confirm the problem, but at the same time, trying to use Windows style path with the `msys2` version of `git` is not a good idea. The `msys2` version of git works best with Unix style paths like `/c/Temp/...`. If this is coming from a tool, I suggest you to switch to a native version of `git` (i.e. [Git For Windows](http://git-for-windows.github.io/), it can be installed proper instead `msys2` if needed). This version is a native Windows version and deals correctly with Windows paths.

Note below, `C:\msys2\tmp` is where `/tmp` is mapped by default, adapt to your system.

Tests using `msys2` git (`/usr/bin/git`):

```
$ mkdir -p /tmp/test_git_msys2
$ cd /tmp/test_git_msys2

$ which git
/usr/bin/git

$ git --version
git version 2.7.2

$ git clone https://github.com/rtyley/small-test-repo.git 'C:/msys2/tmp/test_git_msys2/another'
$ ls
another/    C:/
$ rm -rf *

$ git clone https://github.com/rtyley/small-test-repo.git 'C:\msys2\tmp\test_git_msys2\another'
$ ls
another/
$ rm -rf *
```

Tests using `mingw64` a.k.a Git For Windows version git (`/mingw64/bin/git`):

```
$ mkdir -p /tmp/test_git_mingw64
$ cd /tmp/test_git_mingw64

$ which git
/mingw64/bin/git

$ git --version
git version 2.7.2.windows.1

$ git clone https://github.com/rtyley/small-test-repo.git 'C:/msys2/tmp/test_git_mingw64/another'
$ ls
another/
$ rm -rf *

$ git clone https://github.com/rtyley/small-test-repo.git 'C:\msys2\tmp\test_git_mingw64\another'
$ ls
another/
$ rm -rf *
```

Bottom line, use either Unix path (`/c/Users/...`) or Windows Path (`C:\msys2\tmp`) or use a native `git` package.

Regards,
Matt

**Comment by zdm** [comment]
Thank you for advice.
I tried windows native git and it works without this issue.

**Comment by maoueh** [comment]
@zdm, no problem.

@Alexpux Feel free to close issue is you think it's a `won't fix`.

**Comment by mingwandroid** [comment]
I need to fix this.

**Comment by zdm** [comment]
Is this patch already merged and updated git version accessible via pacman? I see, that issue still present.

**Comment by mati865** [comment]
@zdm You can check it. Compare your git version with pkgbuild in git repo.
```
$ pacman -Ss |grep "/git"
msys/git 2.8.1-1 (VCS)
```
Pkgbuild in git repo is 2.8.1-2.

**Comment by mingwandroid** [comment]
If you want the fix before it hits the repos, then build it for yourself.

It's probably a lot easier than you imagine.

**Comment by zdm** [comment]
Hi.
FYI, today git was updated to msys/git 2.8.1-5 (VCS) [installed], but issue still present.

**Comment by mingwandroid** [comment]
I had fixed it for Windows-normal paths (using backslashes), but not for Windows-mixed paths (using slashes). Now it should work with either.

**Comment by EchoOfSoul** [comment]
[warning] git man installation related.
`pacman -S git`
>...
warning: could not get file information for usr/share/man/man3/Git::I18N.3pm.gz
warning: could not get file information for usr/share/man/man3/Git::SVN::Editor.3pm.gz
warning: could not get file information for usr/share/man/man3/Git::SVN::Fetcher.3pm.gz
warning: could not get file information for usr/share/man/man3/Git::SVN::Memoize::YAML.3pm.gz
warning: could not get file information for usr/share/man/man3/Git::SVN::Prompt.3pm.gz
warning: could not get file information for usr/share/man/man3/Git::SVN::Ra.3pm.gz
warning: could not get file information for usr/share/man/man3/Git::SVN::Utils.3pm.gz
....
success installed with warning

look at those dir `usr/share/man/man3/Git*` have two `underscore` should it be one `_` or `colon ::`?

`thanks`
